File: src/main/java/com/nir/beans/MethodInfoComponentImpl.java
package com.nir.beans;

import com.nir.utils.math.method.MethodInfoJsonPojo;

import java.util.Collection;
import java.util.List;
import java.util.stream.Collectors;

public class MethodInfoComponentImpl implements MethodInfoComponent {

    private List<JacksonComponent<MethodInfoJsonPojo>> jacksonComponents;

    public MethodInfoComponentImpl(List<JacksonComponent<MethodInfoJsonPojo>> jacksonComponents) {
        this.jacksonComponents = jacksonComponents;
    }

    public List<MethodInfoJsonPojo> getAll() {
        return jacksonComponents.stream().map(JacksonComponent<MethodInfoJsonPojo>::readAll).flatMap(Collection::stream).collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/nir/beans/method/MethodInfoComponentImpl.kt
package com.nir.beans.method

import com.nir.beans.method.generalized.MethodInfoJsonPojo
import com.nir.beans.reaction.parsing.JacksonComponent
import java.util.stream.Collectors

class MethodInfoComponentImpl(private val jacksonComponents: List<JacksonComponent<out MethodInfoJsonPojo>>) :
    MethodInfoComponent {
    override val getMethods: List<MethodInfoJsonPojo>
        get() = jacksonComponents.stream()
                .map { obj: JacksonComponent<out MethodInfoJsonPojo> -> obj.readAll() }.flatMap { obj: List<MethodInfoJsonPojo> -> obj.stream() }
                .collect(Collectors.toList())
}